lib/Dialect/ESI/runtime/cpp/include/esi/Services.h

```diff
@@ -48,6 +49,18 @@ class Service {
 
   virtual std::string getServiceSymbol() const = 0;
 
+  /// Create a "child" service of this service. Used when a service already
+  /// exists in the active services table, but a new one wants to replace it.
```

Comment:
I think this is a bit unclear... does 'a service' here imply that a 'child service' _of the same type_ wants to replace it? but then, below, it says that the child service needs to _use_ the parent service...

Reply:
It doesn't necessarily have to be -- it's entirely up to the `getChildService` method. I've clarified the code doc.